### PR TITLE
pretty-print: add pretty print

### PR DESCRIPTION
**Pull request by AsadullahFarooqi**
haven't tested these prints because I haven't seen any query example

### CODE REVIEW

**Comment by AsadullahFarooqi** [comment]
this PR will be rebased after this PR https://github.com/eto-ai/lance/pull/468

**Comment by AsadullahFarooqi** [comment]
Screenshot from 2023-01-29 01-00-52
The PR is ready IMO @changhiskhan @eddyxu 
but I think https://github.com/eto-ai/lance/issues/477 issue is genuine the limit can't go over 1024 and even if i put some lower limit it still panics rather than graceful ending

**Comment by AsadullahFarooqi** [comment]
just noticed the commit msg pattern you guys use. let me quickly update that

**Comment by eddyxu** [comment]
Since we included many `arrow-*` submodules, curious about what would be the difference in the final binary size to include `arrow`.

**Comment by AsadullahFarooqi** [comment]
> Since we included many `arrow-*` submodules, curious about what would be the difference in the final binary size to include `arrow`.

by binary you mean just the `lq` bin or anything else as well?

File: rust/src/bin/lq.rs [hunk 1]
```diff
@@ -115,7 +116,19 @@ async fn main() -> Result<()> {
...
             let stream = scanner.try_into_stream().await.unwrap();
             let batch: Vec<RecordBatch> = stream.take(1).try_collect::<Vec<_>>().await.unwrap();
-            println!("{:?}", batch);
+
+            // print the summary of data total rows and cols
+            let mut total_num_of_rows: usize = 0;
+            let mut total_num_of_cols: usize = 0;
+            for b in batch.iter() {
+                total_num_of_rows += b.num_rows();
+                total_num_of_cols += b.num_columns();
```
Comment (eddyxu) [inline rust/src/bin/lq.rs:125]: The columns could be the same amount batches. We could just print num of columns from the first batch.

For the number of rows, let's wait for #495 to get in?

The rest LGTM

**Review by eddyxu — APPROVED** [review]

File: rust/src/bin/lq.rs (hunk 1 above)
Comment (AsadullahFarooqi) [reply rust/src/bin/lq.rs]: done

Comment (AsadullahFarooqi) [reply rust/src/bin/lq.rs]: you are right the cols would be same from the 1st batch

**Comment by eddyxu** [comment]
Thanks @AsadullahFarooqi for your contribution.

**Pull request merged**